### PR TITLE
feat(reconcile): API endpoint + UI implementation to trigger reconcile from the UI + controller fix for infinite loops

### PR DESCRIPTION
**Pull request by ramonvermeulen**
This feature adds a `POST /apis/magosproject.io/v1alpha1/workspaces/{namespace}/{name}/reconcile` endpoint to the API that triggers a manual reconciliation by applying the `magosproject.io/reconcile-request` annotation to the Workspace CR. 

The UI now includes a Reconcile button, which is disabled while the workspace is already reconciling. On the controller side, with this PR manual reconciliations do not affect the `nextReconcileAt` timestamp and cannot cause infinite loops. There is also a chainsaw test added to mimic exactly that scenario. If a manual trigger happens right before a scheduled reconcile, the scheduled cycle simply skips one time because the workspace is already busy. Also a fix has been implemented so that when a user changes the reconcile interval of a workspace, while the workspace already had a `nextReconcileTime` planned, the new interval will directly take effect instead of the need to wait a whole new cycle.

This PR also includes several UI refactors to improve code readability and type safety (biggest chunk of the diff).

### CODE REVIEW

File: api/internal/service/workspace_service.go [hunk 1]
```diff
@@ -94,3 +102,35 @@ func (s *workspaceService) Get(_ context.Context, namespace, name string) (*apiv
...
+			},
+		},
+	}
+
+	patchBytes, err := json.Marshal(patch)
+	if err != nil {
+		return nil, err
+	}
+
+	workspace, err := s.client.MagosprojectV1alpha1().Workspaces(namespace).Patch(
```
Comment (ramonvermeulen) [inline api/internal/service/workspace_service.go:120]: When we know from what stages you can start a reconcile, we could maybe fetch the CR and check what stage it is currently in before proceeding with the patch? Can also do this in a follow-up PR. I am not entirely sure, but I assume these stages are `"Applied", "Failed", "Idle"`?

File: internal/controller/workspace/workspace_controller.go [hunk 2]
```diff
@@ -263,18 +264,12 @@ func (r *WorkspaceReconciler) handleDeletion(ctx context.Context, workspace *v1a
 // cleaned up by Step 2. The approval annotation is deliberately excluded from
 // the hash so that approving a plan does not invalidate the existing Plan Job.
 //
-// We also fold the reconcile-request annotation into the hash so that setting
-// that annotation (a manual "re-run" trigger) forces a new plan/apply cycle
-// even when the spec itself hasn't changed.
+// Transient control annotations such as reconcile-request are also excluded.
```
Comment (bschaatsbergen) [inline internal/controller/workspace/workspace_controller.go]: Could you, in the comment, expand on what a “transient control annotation” is in this context, and what it means for it to be excluded?

Comments like this tend to outlive the code’s original context. A bit more detail here helps someone new build the right mental model without having to trace through the controller.

File: internal/controller/workspace/workspace_controller.go [hunk 3]
```diff
@@ -263,18 +264,12 @@ func (r *WorkspaceReconciler) handleDeletion(ctx context.Context, workspace *v1a
...
+// They may trigger a fresh cycle, but they must not change job identity or a
```
Comment (bschaatsbergen) [inline internal/controller/workspace/workspace_controller.go]: This comment feels a bit vague, especially sitting directly above the hashing logic. I lean to suggest removing it to avoid adding ambiguity here. WDYT?

File: internal/controller/workspace/workspace_controller.go [hunk 4]
```diff
@@ -296,6 +291,25 @@ func (r *WorkspaceReconciler) getSyncInterval(ws *v1alpha1.Workspace) time.Durat
 	return DefaultReconciliationInterval
 }
 
+func computeNextScheduledReconcileTime(existing *metav1.Time, interval time.Duration) metav1.Time {
```
Comment (bschaatsbergen) [inline internal/controller/workspace/workspace_controller.go]: I agree with adding this to avoid staggering workspaces on force reconcile.

That said, I think this makes `r.getSyncInterval` redundant. Right now it’s used as the requeue return value, but `computeNextScheduledReconcileTime` encapsulates more of the actual scheduling logic. I’d prefer we standardize on that as the single source of truth for requeue timing.

Keeping both around introduces a foot-gun: it’s easy to accidentally requeue using `r.getSyncInterval` and bypass the newer logic. I’d lean toward removing `r.getSyncInterval` by moving its logic into `computeNextScheduledReconcileTime`. WDYT?

File: test/chainsaw/tests/workspace/manual-reconcile-preserves-schedule/chainsaw-test.yaml
Comment (bschaatsbergen) [inline test/chainsaw/tests/workspace/manual-reconcile-preserves-schedule/chainsaw-test.yaml:1]: This feels quite complex 😄

**Review by bschaatsbergen — COMMENTED** [review]
Thanks for working on this! I’ve left a few style-related comments. General nit: I’d like to see the comments on the newly introduced code to be more in line with the existing style and tone.

File: test/chainsaw/tests/workspace/manual-reconcile-preserves-schedule/chainsaw-test.yaml
Comment (ramonvermeulen) [reply test/chainsaw/tests/workspace/manual-reconcile-preserves-schedule/chainsaw-test.yaml:1]: I know, I was also hesitating to add this but I think these kind of edge-case tests are quite important. Same story for the [reconcile-interval-change-takes-effect-immediately/chainsaw-test.yaml‎](https://github.com/magosproject/magos/pull/34/changes#diff-c5bc0dba038d3532c9e0d901406542a0f5e21af532100c36d1b48149db9c4e6e) case. This is quite specific behavior that you expect from the controller that you want to somehow capture and pin to prevent regression while making other changes.

I know ideally you use the least amount of `script` sections in the chainsaw tests as possible, but I couldn't figure out another way to replicate the scenario that I had in mind. I still tried to explain in comments what and why it is done that way, but we could also decide to remove them entirely.

File: internal/controller/workspace/workspace_controller.go (hunk 3 above)
Comment (ramonvermeulen) [reply internal/controller/workspace/workspace_controller.go]: I agree, removed the entire part because it only makes it more confusing. The getSpecHash just doesn't use the WorkspaceReconcileRequestAnnotation anoymore alongside the hash. No need to explain in a comment why we don't do that.

**Review by bschaatsbergen — APPROVED** [review]
